### PR TITLE
Square Payments: Create landing page

### PR DESCRIPTION
**Pull request by WBerredo**
Related to https://github.com/Automattic/dotcom-forge/issues/7781
Related to https://github.com/Automattic/dotcom-forge/issues/7785

## Proposed Changes

* Create a landing page to be shown on the `/start-with/square-payments`
* Update page layout to mach the design in figma
* Add a CTA to redirect to the store creation endpoint 
* Send a track event for page view and CTA click

PS: Mobile visualization will be handled in a follow-up PR

## Why are these changes being made?

As part of the Partner Aware Onboarding project to shows users a page to about the partnership before starting a store with those definitions selected.

## Testing Instructions

* Go to `/start-with/square-payments` in a logged in and logged out session
* Check if the layout matches the pictures below
* Click on the CTA and check if you are redirected to `/setup/entrepreneur/start?partnerBundle=square`
* Check on tracks if the events are being properly recorded

| Logged out  | Logged in |
| ------------- | ------------- |
|CleanShot 2024-06-20 at 11 00 34@2x|CleanShot 2024-06-20 at 11 02 06@2x|

### CODE REVIEW

File: client/start-with/style.scss [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+					font-size: 79px;
+					line-height: 105%;
+				}
+
+				.subtitle {
+					color: #000;
+					font-family: $font-sf-pro-display;
+					font-size: $font-body;
+					font-style: normal;
+					line-height: 150%;
```
Comment (epeicher) [inline client/start-with/style.scss]:
```suggestion
					line-height: 162.5%;
```
Very minor change, just to match the design

File: client/start-with/style.scss [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+					width: 530px;
```
Comment (epeicher) [inline client/start-with/style.scss:59]:
```suggestion
					width: 530px;
					margin-bottom: 0;
```
This is to remove some margin that is being added in the `p` element

**Review by epeicher — APPROVED** [review]
Great work, I have checked, and it matches the design for both logged-in and logged-out versions. I have also tested the CTA redirection and track event, thanks for adding it.

I have added a couple of minor (non-blocking) comments, but this can be :shipit:

File: client/start-with/style.scss (hunk 2 above)
Comment (WBerredo) [reply client/start-with/style.scss:59]: Added at https://github.com/Automattic/wp-calypso/pull/91993/commits/d358237aa25b9bbc119ae533f571480c91323d6a

File: client/start-with/style.scss (hunk 1 above)
Comment (WBerredo) [reply client/start-with/style.scss]: Fixed at https://github.com/Automattic/wp-calypso/pull/91993/commits/d358237aa25b9bbc119ae533f571480c91323d6a

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/15101213

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/15101213).

**Pull request merged**